### PR TITLE
ENH Remove warnings, add db validation for AbricateNormalizer, add file being normalized in CLI msg.

### PR DESCRIPTION
**Pull request by Vedanth-Ramji**

### CODE REVIEW

File: argnorm/cli.py [hunk 1]
```diff
@@ -30,5 +30,5 @@ def main():
         )
 
     prop_unmapped = ((result.ARO == 'ARO:nan').sum() + result.ARO.isna().sum()) / result.shape[0]
-    print(f'{round(1 - prop_unmapped, 3):.2%} args mapped.')
+    print(f'{args.output}:', f'{round(1 - prop_unmapped, 3):.2%} ARGs mapped.')
```
Comment (luispedro) [inline argnorm/cli.py:33]: This is not really related to removing unnecessary warnings, though

File: argnorm/normalizers.py [hunk 2]
```diff
@@ -194,6 +172,11 @@ def preprocess_ref_genes(self, ref_genes):
 
 class AbricateNormalizer(BaseNormalizer):
     def __init__(self, database=None, is_hamronized=False) -> None:
+        try:
+            database in ['ncbi', 'deeparg', 'resfinder', 'sarg', 'megares', 'argannot']
```
Comment (luispedro) [inline argnorm/normalizers.py:176]: This doesn't do anything. SHould be

```python
if database not in ...:
   raise ...
```

Was there a test for this functionality?

**Review by luispedro — CHANGES_REQUESTED** [review]

File: argnorm/cli.py (hunk 1 above)
Comment (Vedanth-Ramji) [reply argnorm/cli.py:33]: Without the warnings it got a little difficult to see which file was getting normalized. I'd only see 'x% ARGs mapped'. Previously the warnings gave some idea as to which database was being used, but I wanted to make it clear for which file the normalization occurred.

Comment (luispedro) [reply argnorm/cli.py:33]: This makes sense. It should be mentioned in the commit message

Comment (Vedanth-Ramji) [reply argnorm/cli.py:33]: Done!

File: tests/test_normalizers.py [hunk 3]
```diff
@@ -33,25 +34,33 @@ def test_deeparg_normalizer(hamronized):
...
 
-    pd.testing.assert_frame_equal(normed, golden_file)
+        pd.testing.assert_frame_equal(normed, golden_file)
 
-@pytest.mark.parametrize('database', ['argannot', 'megares', 'ncbi'])
+@pytest.mark.parametrize('database', ['argannot', 'megares', 'ncbi', 'sarg', 'random_db'])
 def test_abricate_normalizer_raw(database):
-    normalizer = argnorm.AbricateNormalizer(database=database, is_hamronized=False)
-    input_path = f'./examples/raw/abricate.{database}.tsv'
+    try:
```
Comment (luispedro) [inline tests/test_normalizers.py:53]: Won't this hide errors in the testing?

You should be testing that `random_db` triggers an error (by explicitly failing if it doesn't)

Comment (luispedro) [inline tests/test_normalizers.py:53]: Won't this hide errors in the testing?

You should be testing that random_db triggers an error (by explicitly failing if it doesn't)

File: tests/test_normalizers.py [hunk 4]
```diff
@@ -33,25 +34,33 @@ def test_deeparg_normalizer(hamronized):
...
     pd.testing.assert_frame_equal(normed, golden_file)
 
-@pytest.mark.parametrize('database', ['argannot', 'megares', 'ncbi', 'resfinder'])
+@pytest.mark.parametrize('database', ['argannot', 'megares', 'ncbi', 'resfinder', 'random_db'])
 def test_abricate_normalizer_hamronized(database):
-    normalizer = argnorm.AbricateNormalizer(database=database, is_hamronized=True)
-    input_path = f'./examples/hamronized/abricate.{database}.tsv'
+    try:
+        normalizer = argnorm.AbricateNormalizer(database=database, is_hamronized=True)
+    except:
```
Comment (luispedro) [inline tests/test_normalizers.py:41]: `try`/`except` is generally bad. Be more explicit

Comment (luispedro) [reply tests/test_normalizers.py:41]: but see below. I don't think this is what you want at all

**Review by luispedro — CHANGES_REQUESTED** [review]

File: tests/test_normalizers.py (hunk 3 above)
Comment (Vedanth-Ramji) [reply tests/test_normalizers.py]: I have added two separate tests using `pytest.raises(Exception)`.
